src/main/java/de/hs_mannheim/informatik/ct/controller/PrintOutController.java

```diff
@@ -68,45 +77,35 @@
 
     @GetMapping(value = "/rooms")
     public String getRoomPrintoutList(Model model) {
-        model.addAttribute("buildings", buildingService.getAllBuildings());
         return "rooms/roomPrintout";
     }
 
-    @GetMapping(value = "/rooms/{building}")
-    public DeferredResult<ResponseEntity<byte[]>> getRoomPrintout(
-            @PathVariable(value = "building") String building,
-            HttpServletRequest request) {
-        val roomsInBuilding = buildingService.getAllRoomsInBuilding(building);
-        if (roomsInBuilding.size() == 0) {
-            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Building not found or empty");
-        }
 
-        val outFileName = String.format("Gebäude %s.docx", building);
+    //    TODO IMPORTANT this side can be access without being logged in
+    @GetMapping(value = "/rooms/download", produces = "application/zip")
+    public void getRoomPrintout(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+        //setting headers
+        response.setStatus(HttpServletResponse.SC_OK);
+        response.addHeader("Content-Disposition", "attachment; filename=\"allRoomNotes.zip\"");
```

Comment:
How are the privileged printouts downloaded? I guess you need a second method for that, just make sure to reuse as much code as possible. Probably by extracting most of the code into a private method that you call from both of the request methods.